cvat/apps/engine/task.py

```diff
@@ -354,6 +365,8 @@ def _create_thread(tid, data):
         raise NotImplementedError("Adding more data is not implemented")
 
     upload_dir = db_task.get_upload_dirname()
+    if data['remote_files']:
+        data['client_files'] = _download_data(data['remote_files'], upload_dir)
```

Comment:
It looks like a hack. It works but please don't assign client_files == remote_files. It is better to change *validate* function.

Reply:
The download procedure has to be moved to validate function also. Is it OK?

Reply:
A problem is that the upload_dir is not known in validate function. I have no idea how to change validate function.

Reply:
@zliang7 , why you cannot write `data['remote_files'] = _download_data(....)` and use `data['remote_files']`?